src/java/org/apache/cassandra/net/Message.java

```diff
@@ -711,15 +708,12 @@ private Serializer()
 
         public <T> void serialize(Message<T> message, DataOutputPlus out, int version) throws IOException
         {
-            if (version >= VERSION_40)
-                serializePost40(message, out, version);
-            else
-                serializePre40(message, out, version);
+            serializePost40(message, out, version);
```

Comment:
All the methods that were named `*Post40` can be renamed or inlined into the called methods, since they don't have a `*Pre40` counterpart anymore.

Reply:
i thought of that… but what about making forward-merge simpler? 

any harm in keeping it for another release cycle, cleaning it later?

Reply:
If we leave it for later we will still be complicating merge. In general I prefer simpler code over simpler merges, and since the ticket is about cleaning up it seem that renaming the methods is in scope. I wouldn't be against doing it later in a separate ticket though, but I think in that case we should probably leave some TODOs about that so it's less confusing for readers. wdyt?

Reply:
see commit: https://github.com/apache/cassandra/pull/2506/commits/c786e6687d2ecb810fd5e22b9d34d54ca9076438

Reply:
I see that the methods have already inlined, which is what impacts merge the most. The remaining *Impl methods aren't implementations of other methods, I think, but private utility methods. So instead of renaming with `s/Post40/Impl/` we can just remove the `Post40`/`Impl` suffix. The methods I mean are:
* serializeHeaderImpl
* deserializeHeaderImpl
* skipHeaderImpl
* inferMessageSizeImpl
* extractParamsSizeImpl

There shouldn't be a lot of difference between the `s/Post40/Impl/` rename and just dropping the suffix when merging forward.

Reply:
💯